src/main/java/com/cvsgo/repository/ProductRepository.java

```diff
@@ -1,10 +1,19 @@
 package com.cvsgo.repository;
 
 import com.cvsgo.entity.Product;
+import jakarta.persistence.LockModeType;
+import java.util.Optional;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Lock;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
 
 public interface ProductRepository extends JpaRepository<Product, Long>, ProductCustomRepository {
 
+    @Lock(LockModeType.OPTIMISTIC)
+    @Query(value = "select p from Product p where p.id = :id")
+    Optional<Product> findByIdForUpdate(@Param("id") Long id);
```

Comment:
for update는 pessimistic lock을 걸 때 사용하는 SQL문이라서 optimistic lock을 사용하는 이 메소드는 다른 이름을 사용하는 것이 좋을 것 같습니다!

Reply:
```suggestion
    @Lock(LockModeType.OPTIMISTIC)
    @Query(value = "select p from Product p where p.id = :id")
    Optional<Product> findByIdWithOptimisticLock(@Param("id") Long id);
```
그렇군요,,, 해당 메서드 이름 혹시 어떠세요??

Reply:
괜찮은 것 같습니다!